contracts/DLCManager.sol

```diff
@@ -239,39 +238,18 @@ contract DLCManager is
             if (!hasRole(APPROVED_SIGNER, attestorPubKey)) {
                 revert InvalidSigner();
             }
-            _checkSignerUnique(seenSigners, attestorPubKey);
-            seenSigners[i] = attestorPubKey;
+            _checkSignerUnique(attestorPubKey, prefixedMessageHash);
         }
     }
 
-    /**
-     * @notice  Checks for duplicate values in an array.
-     * @dev     Used to check for duplicate signatures.
-     * @param   signatures  Array of signatures.
-     * @return  bool  True if there are duplicates, false otherwise.
-     */
-    function _hasDuplicates(
-        bytes[] memory signatures
-    ) internal pure returns (bool) {
-        for (uint i = 0; i < signatures.length - 1; i++) {
-            for (uint j = i + 1; j < signatures.length; j++) {
-                if (keccak256(signatures[i]) == keccak256(signatures[j])) {
-                    return true;
-                }
-            }
-        }
-        return false;
-    }
-
     function _checkSignerUnique(
-        address[] memory seenSigners,
-        address attestorPubKey
-    ) internal pure {
-        for (uint256 j = 0; j < seenSigners.length; j++) {
-            if (seenSigners[j] == attestorPubKey) {
-                revert DuplicateSigner(attestorPubKey);
-            }
+        address attestorPubKey,
+        bytes32 messageHash
+    ) internal {
+        if (_seenSigners[attestorPubKey][messageHash]) {
+            revert DuplicateSigner(attestorPubKey);
```

Comment:
Maybe put a comment here to explain that the messageHash is globally unique? Or if it's globally unique, why do we even need to use the attestorPubKey indexing at all?

isn't this going to grow forever? what if it gets really big?

Reply:
Only this attestor’s signature for this message is globally unique, which is why i chose to store that. 
Other attestors sign the same message, and same attestors sign different messages, so only this particular combination of pubkey-hash is globally unique.

Maximum size of a mapping is 2^256 members :)

Reply:
that's a lot of members :) SO it's better to let it grow forever than to clear out the map?

Reply:
yes!